### ISSUE TITLE
When build the app with highcharts are not working...

### ISSUE THREAD

**Issue by chandumunna** [JoeKuan/Highcharts_Sencha#25]
in app.json file use the sencha-touch-all.js then build the app  shown error undefined not a function.....

my code is:

{
                xtype:'highchart',
               
                series : [{
                    type : 'column',
                    dataIndex : 'yesterday',
                    name : 'Yesterday'
                }, {
                    type : 'column',
                    dataIndex : 'today',
                    name : 'Today'
                }],

                xField : 'time',
                chartConfig : {
                    chart : {
                        zoomType : 'x'
                    },
                    title : {
                        text : 'Bar chart for landscape orientation and Column chart for Portrait orientation',
                        x : -20 //center
                    },
                    subtitle : {
                        text : 'Random Value',
                        x : -20
                    },
                    xAxis : [{
                        title : {
                            text : 'Time',
                            margin : 20
                        },
                        labels : {
                            rotation : 270,
                            y : 35,
                            formatter : function () {
                                var dt = Ext.Date.parse (parseInt (this.value) / 1000, "U");
                                if (dt) {
                                    return Ext.Date.format (dt, "H:i:s");
                                }
                                return this.value;
                            }

                        }
                    }],
                    yAxis : {
                        title : {
                            text : 'Temperature'
                        },
                        plotLines : [{
                            value : 0,
                            width : 1,
                            color : '#808080'
                        }]
                    },
                    tooltip : {
                        formatter : function () {
                            var dt = Ext.Date.parse (parseInt (this.x) / 1000, "U");
                            return 'At <b>' + this.series.name + '</b>' + Ext.Date.format (dt, "H:i:s") + ',<br/>temperature is : ' + this.y;
                        }

                    },
                    legend : {
                        layout : 'vertical',
                        align : 'right',
                        verticalAlign : 'top',
                        x : -10,
                        y : 100,
                        borderWidth : 0
                    },
                    credits : {
                        text : 'joekuan.wordpress.com',
                        href : 'http://joekuan.wordpress.com',
                        style : {
                            cursor : 'pointer',
                            color : '#707070',
                            fontSize : '12px'
                        }
                    }
                },
                listeners:{
                    initialize:function(){
                        console.log(this.series)
                        var myStore = Ext.create("Ext.data.Store", {
                            autoLoad : false,
                            storeId: 'temptemperature',
                            fields : [{
                                name : 'time',
                                type : 'string'
                            }, {
                                name : 'yesterday',
                                type : 'float'
                            }, {
                                name : 'today',
                                type : 'float'
                            }],
                            proxy: {
                                type : 'ajax',
                                url : 'temp_example.php',
                                extraParams: { summary: 0 },
                                reader : {
                                    type : 'json',
                                    rootProperty : 'rows'
                                }
                            }
                        });
                       
                        console.log(myStore)
                        this.bindStore(myStore,true)
                        myStore.load()
                    }
                }
            }

shown error: 

![screen shot 2014-06-06 at 3 04 54 pm](https://cloud.githubusercontent.com/assets/3447677/3198739/102f399c-ed5e-11e3-8836-6d99851d8e9f.png)

**Comment by bertonoronha** [comment]
Hello there,

I would also be interested in that solution. If you could forward the email for me as well I would appreciate it.

alberto.noronha@uni.lu

Best regards,
Alberto

**Comment by salorob** [comment]
Hi There,

I'm also experiencing that problem.  I managed to solve it by adding a script reference to http://code.highcharts.com/highcharts.js, however i'm getting another error: 
Uncaught TypeError: undefined is not a function, it seems around titleSetter:function(a){ ...

How can this be resolved?

**Comment by JoeKuan** [comment]
I can build an application with sencha cmd tool and it works fine. I have documented it in a blog. 

https://joekuan.wordpress.com/2015/02/28/building-extjshighcharts-application-with-sencha-cmd-tool/

Let me know whether this helps. If not, then please send me the code.

**Comment by salorob** [comment]
Hi JoeKuan,

Thanks!

I managed to get this working after all, but dit you see my other post regarding DateTime objects from sencha touch, where de datetimelabelformat is not working?

Met vriendelijke groet,

[cid:16CD9A3C-DE21-455B-93F6-5ABF1E4173B3]

Robert Salomons
r.salomons@ropesolutions.nl
Tel: 038 - 777 9830

**Comment by JoeKuan** [comment]
I got another sencha Cmd 4 issue need to investigate this weekend. Then I will move on to this one. I will have a look. Sorry for the delay again.

**Comment by salorob** [comment]
Hi JoeKuan,

I must say, that when googling around I found out that somebody said you needed to specify BOTH the xField and the yField in the Series: {} , after I did this things started working.

Also, in my store/model I had a date field that was fed by ISO8601 date format, I had to create a field with a converter function() that returned the .getTime() of the object. When I started using that field, things started working. (when using the field that held the Ext.Date object itsself, the date in highcharts seem to start in 1970, in stead of 2015. (is that a known issue ?)

Also, to be curious; are you the creator of Highcharts or only of the Sencha Adapter of it ?

Eitherway; great work :P

Met vriendelijke groet,

[cid:7203FB7A-356D-4453-A570-41428AB73081]

Robert Salomons
r.salomons@ropesolutions.nl
Tel: 038 - 777 9830
